### ISSUE TITLE
B4 Definitions of gcis:TemporalExtents and gcis:SpatialExtents

### ISSUE THREAD

**Issue by xgmachina** [USGCRP/gcis-ontology#72]
In http://data.globalchange.gov/gcis.owl#TemporalExtents and  http://data.globalchange.gov/gcis.owl#SpatialExtents, should the domains and ranges explicitly call out “datasets” given the descriptions of these terms?
Comment: Might be good to see ISO’s Extents to reduce reinventing the wheel. more specifically the geographical and temporal extents. ISO also distinguishes with vertical extents too.
e.g. https://ontohub.org/socop/ISO19115.owl
GCIS currently pulling OAI-PMH feeds from several of the NASA DAACs. There is discussion about ingesting from GeoPortal, which follows ISO.

**Comment by xgmachina** [comment]
`gcis:TemporalExtents` and `gcis:SpatialExtents` are both classes, not properties, so I think the original update suggestion focuses more on the term 'dataset' in the literal definition of those two classes. If that is the case then it could be easy to make updates.

**Comment by rewolfe** [comment]
I like the idea of using ISO in this area.

**Comment by justgo129** [comment]
ISO it is.  @xgmachina please feel free to proceed.

**Comment by zednis** [comment]
I think we should do some research on the 'preferred' ISO 19115 OWL encoding.  There are a couple out there, and if OGC has a preferred encoding I think we should use that.

Simon Cox is heavily involved in RDF/OWL encodings for OGC/ISO standards has an OWL encoding of ISO 19115 at http://def.seegrid.csiro.au/static/isotc211/iso19115/2003/extent.ttl.  I will ask him what the current preferred encoding is.

**Comment by zednis** [comment]
I have sent a message to Simon and will update this thread with his response.

**Comment by zednis** [comment]
Here are Simon's responses to my questions on ISO 19115 extent encodings in OWL.

SZ:  Do you know if there is a preferred encoding of ISO 19115 extents in OWL?

> This is on the agenda in the SDWWG (OGC/W3C joint working group).
> My prediction is that there will be more than one option, but fewer than six :smile: 
> For spatial objects the tendency is towards embedding a micro-format, like WKT or GeoJSON or even GML, like GeoSPARQL does.
> 
> The truth is that coordinates are complex objects, which couple more than one item of information together each of which is essentially meaningless without the others. We recognise this for time encodings (e.g. 2015-07-29T08:43:00.00+10:00 is actually seven pieces of information encoded in a microformat) and need to bite the bullet for other geometries.

SZ: So far I have found https://ontohub.org/socop/ISO19115.owl 

> This was Luis Bermudez work from many many years ago. Has been superseded by ISO 19150-2-based conversions like …

SZ:  and http://def.seegrid.csiro.au/static/isotc211/iso19115/2003/extent.ttl

> … but no-one in their right mind would use these in practice. They are too literally based on the UML.
> I have had some conversations on ‘what are such UML-y OWL models good for’ and AFAICT the best answer is ‘they provide URIs for every element in the UML model, which can then be used in alignment and mapping graphs’ (this is the answer I proposed). INSPIRE might slog away adding it to their edifice, but I think even they are having second thoughts.
> 
> I would stick with GeoSPARQL, until the SDWWG make their ruling.

**Comment by justgo129** [comment]
With regard to the specific question:
In http://data.globalchange.gov/gcis.owl#TemporalExtents and http://data.globalchange.gov/gcis.owl#SpatialExtents, should the domains and ranges explicitly call out “datasets” given the descriptions of these terms?

a) I suggest removing "of a dataset" from the rdfs:comment of gcis:TemporalExtents:
    rdfs:comment "The time information, including that of the earliest and last observations, of a dataset." ;

b) I suggest replacing "the data or results" with "a resource" in the rdfs:comment for gcis:SpatialExtents.
    rdfs:comment "A bounding box describing the northern, southern, western, and eastern limits of the data or results." ;

Within the sole context of this ticket, is this an acceptable proposal?

I apologize for any confusion; the sole rationale was the mentioning of a "dataset" in the rdfs:comment for the aforementioned terms.

**Comment by zednis** [comment]
@justgo129 said

> should the domains and ranges explicitly call out “datasets” given the descriptions of these terms?

I am not sure which domains and ranges you are referring to.  The extent classes are in the domain of properties we have defined to help characterize extents (start/end properties, latitude/longitude properties).  I do not believe these classes should be in the range of any current properties.

If I remember correctly we agreed to move towards using dublin core's `dcterms:spatial` and `dcterms:temporal` properties, correct? (#10, #11) and removed `gcis:hasSpatialExtents` and `gcis:hasTemporalExtents`.

I believe the HTML you are referencing is out-of-date.

The range of `dcterms:spatial` is `dcterms:Location` and the range of `dcterms:temporal` is `dcterms:PeriodOfTime`.  It is up to us to determine how best to characterize instances of `dcterms:PeriodOfTime` and `dcterms:Location` using other community vocabularies or our own.

The instances of `dcterms:Location` and `dcterms:PeriodOfTime` in the RDF can also have other types, including GCIS-defined types, so we can keep the extent classes we have - but I do not think it is necessary for us to do so.

I say we keep them but give them broad definitions that are compatible with the definitions of PeriodOfTime and Location from dublin core.  We could even use the same definition as the compatible terms from dublin core.

`dcterms:Location` - "A spatial region or named place."
`dcterms:PeriodOfTime` - "An interval of time that is named or defined by its start and end dates."

Possible broad definitions of gcis extent classes.  I removed the reference to 'resource' because the semantics of the connection between the spatial or temporal information and the resource with that information is provided by the `dcterms:spatial` and `dcterms:temporal` properties.

`gcis:SpatialExtents` - "A spatial region defined by a bounding box"
`gcis:TemporalExtents` - "An interval of time defined by its start and end dates or datetimes"

and this is how the whole thing would be represented in the instance data

```
<http://data.globalchange.gov/dataset/nasa-nsidcdaac-0042>
  dcterms:temporal [
    a dcterms:PeriodOfTime, gcis:TemporalExtents ;
    gcis:startDate "1987-08-01"^^xsd:date ;
    gcis:endDate "1988-12-31"^^xsd:date ;
  ] ;
  dcterms:spatial [
    a dcterms:Location, gcis:SpatialExtents ;
    gcis:northBoundLatitude "88"^^xsd:float ;
    gcis:southBoundLatitude "-88"^^xsd:float ;
    gcis:eastBoundLongitude "180"^^xsd:float ;
    gcis:westBoundLongitude "-180" ^^xsd:float
  ] .
```

links to publisher guide on using dublin core terms RDF:
- http://wiki.dublincore.org/index.php/User_Guide/Publishing_Metadata#dcterms:temporal
- http://wiki.dublincore.org/index.php/User_Guide/Publishing_Metadata#dcterms:spatial

**Comment by zednis** [comment]
Perhaps we should address this in a separate issue - but did we intend for for the spatial and temporal extent classes to have a plural name?

**Comment by justgo129** [comment]
I'll defer to @rewolfe about the solution of @zednis.  Yes, let's spin off the plural name issue into a second ticket.

**Comment by justgo129** [comment]
Also @zednis, just as a general point of information: with regard to the turtle above (northBoundLatitude, etc.) could we use xsd:float regardless of the object?  i.e., if someone reports "-88" as in the example above  or "-88.0" or "'88.01"?

**Comment by rewolfe** [comment]
Stephan's definitions and the example above look good.  Singular makes more
sense for the extents since I don't think the idea is to provide more than
one spatial or temporal extent.

On Tue, Aug 18, 2015 at 6:34 PM, justgo129 notifications@github.com wrote:

> Also @zednis https://github.com/zednis, just as a general point of
> information: with regard to the turtle above (northBoundLatitude, etc.)
> could we use xsd:float regardless of the object? i.e., if someone reports
> "-88" as in the example above or "-88.0" or "'88.01"?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/issues/72#issuecomment-132376439
> .

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by zednis** [comment]
@justgo129

>  could we use xsd:float regardless of the object? i.e., if someone reports "-88" as in the example above or "-88.0" or "'88.01"?

I am not sure what you are asking.  Are you asking for clarification on what is and is not a valid value for a xsd:float?

**Comment by justgo129** [comment]
Yes.  Can we use a float for a value of "-88" since that's an integer?  What happens if one value is "-88.1" (float) and another is "-88" (integer)?  Would  it therefore be problematic to hard-code "xsd:float"?

**Comment by zednis** [comment]
I believe "-88" is a valid value for a literal of type `xsd:float`, have you seen any documentation that suggests otherwise?

**Comment by justgo129** [comment]
You're right .I just hadn't seen them represented that way.  It seems that C initially handled them differently; which was the source if my confusion.  (My intro to CS was C).  Feel free to proceed.

**Comment by justgo129** [comment]
@zednis, I like your suggestion (in comment beginning "I'm not sure.")  Please feel free to act on it.

**Comment by justgo129** [comment]
Also, let's make "temporalextents" and "spatialextents" as singular per the suggestion of @rewolfe.

**Comment by justgo129** [comment]
Closed #72 on behalf of merged https://github.com/USGCRP/gcis/pull/231.
